### DISCUSSION TITLE
Is there a way to distinguish between inserted values and values added by clickhouse in JSON type ?

### DISCUSSION THREAD

**Discussion by Derianer** [ClickHouse/ClickHouse#62235] [Q&A] [answered]
Hello

Problem is that clickhouse adds fields with values to previously saved objects if I save a new object with additional fields.
For example: 
``` sql
SET allow_experimental_object_type=1;

CREATE TABLE test_json (obj JSON) ENGINE = MergeTree ORDER BY tuple();

INSERT INTO test_json FORMAT JSONEachRow {"obj": {"key1": 1}};

INSERT INTO test_json FORMAT JSONEachRow {"obj": {"key1": 1, "key2": 2}};

SELECT obj FROM test_json FORMAT JSONEachRow SETTINGS output_format_json_named_tuples_as_objects=1;
```
Output:
```
{"obj":{"key1":1,"key2":2}}
{"obj":{"key1":1,"key2":0}}
```

Is there a way to distinguish values that were inserted with the query and values added by clickhouse ? Or maybe there is an option to save such values as NULL?

**Comment by avelychko12** [comment, marked as answer]
Try to use `Object(Nullable('json'))` type instead of `JSON`

**Reply by Derianer** [reply]
it worked, thanks
